include/openthread/platform/ble.h

```diff
@@ -0,0 +1,1013 @@
+    OT_BLE_ADDRESS_LENGTH = 6,
+
+    /**
+     * Time slot duration on PHY layer in microseconds (0.625ms).
+     */
+    OT_BLE_TIMESLOT_UNIT = 625,
+
+    /**
+     * Unit used to calculate connection interval (1.25ms)
+     */
+    OT_BLE_CONN_INTERVAL_UNIT = 2 * OT_BLE_TIMESLOT_UNIT,
+
+    /**
+     * Minimum allowed connection interval in OT_BLE_CONN_INTERVAL_UNIT units (7.5ms).
+     */
+    OT_BLE_CONN_INTERVAL_MIN = 0x0006,
+
+    /**
+     * Maximum allowed connection interval in OT_BLE_CONN_INTERVAL_UNIT units (4s).
+     */
+    OT_BLE_CONN_INTERVAL_MAX = 0x0C80,
+
+    /**
+     * Maximum allowed slave latency in units of connection events.
+     */
+    OT_BLE_CONN_SLAVE_LATENCY_MAX = 0x01F3,
+
+    /**
+     * Minimum allowed connection timeout in units of 10ms (100ms).
+     */
+    OT_BLE_CONN_SUPERVISOR_TIMEOUT_MIN = 0x000A,
+
+    /**
+     * Maximum allowed connection timeout (32s).
+     */
+    OT_BLE_CONN_SUPERVISOR_TIMEOUT_MAX = 0x0C80,
+
+    /**
+     * Unit used to calculate connection supervisor timeout (10ms).
+     */
+    OT_BLE_CONN_SUPERVISOR_UNIT = 16 * OT_BLE_TIMESLOT_UNIT,
+
+    /**
+     * Maximum length of the device name characteristic.
+     */
+    OT_BLE_DEV_NAME_MAX_LENGTH = 248,
+
+    /**
+     * Maximum length of advertising data.
+     */
+    OT_BLE_ADV_DATA_MAX_LENGTH = 31,
+
+    /**
+     * Maximum length of scan response data.
+     */
+    OT_BLE_SCAN_RESPONSE_MAX_LENGTH = 31,
+
+    /**
+     * Minimum allowed interval for advertising packet in OT_BLE_ADV_INTERVAL_UNIT units (20ms).
+     */
+    OT_BLE_ADV_INTERVAL_MIN = 0x0020,
+
+    /**
+     * Maximum allowed interval for advertising packet in OT_BLE_ADV_INTERVAL_UNIT units (10.24s).
+     */
+    OT_BLE_ADV_INTERVAL_MAX = 0x4000,
+
+    /**
+     * Unit used to calculate interval duration (0.625ms).
+     */
+    OT_BLE_ADV_INTERVAL_UNIT = OT_BLE_TIMESLOT_UNIT,
+
+    /**
+     * Minimum allowed scan interval (2.5ms).
+     */
+    OT_BLE_SCAN_INTERVAL_MIN = 0x0004,
+
+    /**
+     * Maximum allowed scan interval (10.24s).
+     */
+    OT_BLE_SCAN_INTERVAL_MAX = 0x4000,
+
+    /**
+     * Unit used to calculate scan interval duration (0.625ms).
+     */
+    OT_BLE_SCAN_INTERVAL_UNIT = OT_BLE_TIMESLOT_UNIT,
+
+    /**
+     * Minimum allowed scan window in OT_BLE_TIMESLOT_UNIT units (2.5ms).
+     */
+    OT_BLE_SCAN_WINDOW_MIN = 0x0004,
+
+    /**
+     * Maximum allowed scan window in OT_BLE_TIMESLOT_UNIT units (10.24s).
+     */
+    OT_BLE_SCAN_WINDOW_MAX = 0x4000,
+
+    /**
+     * Unit used to calculate scan window duration (0.625ms).
+     */
+    OT_BLE_SCAN_WINDOW_UNIT = OT_BLE_TIMESLOT_UNIT,
+
+    /**
+     * BLE HCI code for remote user terminated connection.
+     */
+    OT_BLE_HCI_REMOTE_USER_TERMINATED = 0x13,
+
+    /**
+     * Value of invalid/unknown handle.
+     */
+    OT_BLE_INVALID_HANDLE = 0x0000,
+
+    /**
+     * Maximum size of BLE Characteristic.
+     */
+    OT_BLE_CHARACTERISTIC_MAX_LENGTH = 128,
```

Comment:
Actually, the maximum length of characteristic is 512. The number 128 comes from specific application implementation.